pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -234,11 +280,162 @@ func (r *reconciler) currentLoadBalancerService(ci *operatorv1.IngressController
 	return true, service, nil
 }
 
+// createLoadBalancerService creates a load balancer service.
+func (r *reconciler) createLoadBalancerService(service *corev1.Service) error {
+	if err := r.client.Create(context.TODO(), service); err != nil {
+		return fmt.Errorf("failed to create load balancer service %s/%s: %v", service.Namespace, service.Name, err)
+	}
+	log.Info("created load balancer service", "namespace", service.Namespace, "name", service.Name)
+	return nil
+}
+
+// deleteLoadBalancerService deletes a load balancer service.
+func (r *reconciler) deleteLoadBalancerService(service *corev1.Service, options *crclient.DeleteOptions) error {
+	if err := r.client.Delete(context.TODO(), service, options); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to delete load balancer service %s/%s: %v", service.Namespace, service.Name, err)
+	}
+	log.Info("deleted load balancer service", "namespace", service.Namespace, "name", service.Name)
+	return nil
+}
+
+// updateLoadBalancerService updates a load balancer service.  Returns a Boolean
+// indicating whether the service was updated, and an error value.
+func (r *reconciler) updateLoadBalancerService(current, desired *corev1.Service, platform *configv1.PlatformStatus) (bool, error) {
+	changed, updated, needRecreate := loadBalancerServiceChanged(current, desired, platform)
+	if !changed {
+		return false, nil
+	}
+
+	if needRecreate {
+		log.Info("load balancer scope changed, which requires deleting and recreating the load balancer service on platform %s", platform.Type)
+		foreground := metav1.DeletePropagationForeground
+		deleteOptions := crclient.DeleteOptions{PropagationPolicy: &foreground}
+		if err := r.deleteLoadBalancerService(current, &deleteOptions); err != nil {
+			return false, err
+		}
+		if err := r.createLoadBalancerService(desired); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+
+	if err := r.client.Update(context.TODO(), updated); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// loadBalancerServiceChanged checks if the current load balancer service spec
+// matches the expected spec and if not returns an updated one.
+func loadBalancerServiceChanged(current, expected *corev1.Service, platform *configv1.PlatformStatus) (bool, *corev1.Service, bool) {
+	scopeChanged := false
+	currentAnnotations := current.Annotations
+	if currentAnnotations == nil {
+		currentAnnotations = map[string]string{}
+	}
+	expectedAnnotations := expected.Annotations
+	if expectedAnnotations == nil {
+		expectedAnnotations = map[string]string{}
+	}
+	for name := range InternalLBAnnotations[platform.Type] {
+		if currentAnnotations[name] != expectedAnnotations[name] {
+			scopeChanged = true
+		}
+	}
```

Comment:
At this point, it seems that if the `needsRecreate` determination were moved here and if `scopeChanged && needsRecreate` you could return immediately, bypassing all the irrelevant diffing logic for the case when an update is possible.

Reply:
Hmm, that makes sense.  It would incur a second update if the user modified the scope and another field, but scope is really the only thing we expect users to modify at this time.

Reply:
Latest push implements something along these lines.

Reply:
>Hmm, that makes sense. It would incur a second update if the user modified the scope and another field, but scope is really the only thing we expect users to modify at this time.

This is something I hadn't considered. Given that scenario I would expect the scope change to be batched with all other possible changes. I retract my suggestion

Reply:
I don't feel too strongly about it... but since you retracted the suggestion, I've reverted the change.  * grin *.

Reply:
(But I kept the change where I factored `loadBalancerServiceScopeChanged` out of `loadBalancerServiceChanged`: https://github.com/openshift/cluster-ingress-operator/compare/395121e5f4daa1fb4d3bc4a8145aa0324627dba1..328e6f49ceae326c0ca8e3c7d34535787d00a36f)